test/server/echo.go

```diff
@@ -39,6 +40,9 @@ var (
 	ports     []int
 	grpcPorts []int
 	version   string
+
+	// Record how many attempts have been made using a particular sequence of codes
+	state = make(map[string]int)
```

Comment:
Ed, I worry that expecting the pod to respond in an exact sequence is likely to lead to flaky tests. Kubernetes does not guarantee that pods are not killed/restarted in a long test. This is more true for our test cluster which gets overloaded sometimes and fails to schedule pods on time.